src/System.Reflection/tests/PropertyInfo/PropertyInfo_propertyTests.cs

```diff
@@ -13,230 +13,88 @@ namespace System.Reflection.Tests
 
             Assert.NotNull(pi);
             Assert.NotNull(pi.DeclaringType.Name);
-            Assert.Equal("BaseProperty", pi.DeclaringType.Name);
+            Assert.Equal(typeName, pi.DeclaringType.Name);
         }
 
-
-        //Verify PropertyType for PropertyInfo
-        [Fact]
-        public static void TestPropertyType1()
+        [Theory]
+        [InlineData("DerivedPropertyProperty", typeof(DerivedProperty), "Int32")]
+        [InlineData("BasePropertyProperty", typeof(BaseProperty), "Int32")]
+        [InlineData("MyPropAA", typeof(SampleProperty), "Int16")]
+        public static void TestPropertyType(String propName, Type type, String typeName)
         {
-            string propName = "DerivedProeprtyProperty";
-            PropertyInfo pi = GetProperty(typeof(DerivedProeprty), propName);
-
-            Assert.NotNull(pi);
-            Assert.NotNull(pi.PropertyType);
-            Assert.Equal("Int32", pi.PropertyType.Name);
-        }
-
-        //Verify PropertyType for PropertyInfo
-        [Fact]
-        public static void TestPropertyType2()
-        {
-            string propName = "BasePropertyProperty";
-            PropertyInfo pi = GetProperty(typeof(BaseProperty), propName);
-
-            Assert.NotNull(pi);
-            Assert.NotNull(pi.PropertyType);
-            Assert.Equal("Int32", pi.PropertyType.Name);
-        }
-
-        //Verify PropertyType for PropertyInfo
-        [Fact]
-        public static void TestPropertyType3()
-        {
-            string propName = "MyPropAA";
-            PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
-
+            PropertyInfo pi = GetProperty(type, propName);
 
             Assert.NotNull(pi);
             Assert.NotNull(pi.PropertyType);
-            Assert.Equal("Int16", pi.PropertyType.Name);
-        }
-
-
-        //Verify Name for PropertyInfo
-        [Fact]
-        public static void TestName1()
-        {
-            string propName = "MyPropAA";
-            PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
-
-
-            Assert.NotNull(pi);
-            Assert.Equal(propName, pi.Name);
-        }
-
-        //Verify Name for PropertyInfo
-        [Fact]
-        public static void TestName2()
-        {
-            string propName = "MyPropBB";
-            PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
-
-            Assert.NotNull(pi);
-            Assert.Equal(propName, pi.Name);
+            Assert.Equal(typeName, pi.PropertyType.Name);
         }
 
-
-        //Verify Name for PropertyInfo
-        [Fact]
-        public static void TestName3()
+        [Theory]
+        [InlineData("MyPropAA")]
+        [InlineData("MyPropBB")]
+        [InlineData("MyPropCC")]
+        public static void TestName(String propName)
         {
-            string propName = "MyPropCC";
             PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
 
             Assert.NotNull(pi);
```

Comment:
Do we need all these `Assert.NotNull` checks (other tests make sure that these properties exist). If we do get null back then a NRE is thrown (so the test fails anyway).
If you're keen on keeping the assertion, maybe move it to `GetProperty` to simplify each test